### DISCUSSION TITLE
nushell: mise.nu breaks path completion

### DISCUSSION THREAD

**Discussion by deas** [jdx/mise#3554] [Troubleshooting and bug reports]
**Describe the bug**
Following [nushell configuration](https://mise.jdx.dev/getting-started.html#nushell) breaks path completion

**To Reproduce**
Use a system without default nushell configuration, such that executing `nu` creates  initial configuration files in `~/.config/nushell`. In `nu`, ensure that completion works for binaries.

e.g. I have qemu installed. Hence, typing `qemu-<TAB` show various binary execution completions.

```
❯ | qemu-
qemu-io                     qemu-nbd                    qemu-img                    qemu-pr-helper              
qemu-system-arm             qemu-system-i386            qemu-system-armel           qemu-system-armhf           
qemu-system-amd64           qemu-system-arm64           qemu-system-x86_64          qemu-storage-daemon         
qemu-system-aarch64         qemu-system-x86_64-spice    qemu-system-x86_64-microvm  
```

Now do as per docs:

```
'
let mise_path = $nu.default-config-dir | path join mise.nu
^mise activate nu | save $mise_path --force
' | save $nu.env-path --append
"\nuse mise.nu" | save $nu.config-path --append
```

Start a new `nu` shell process. The completion no longer works. 

**Expected behavior**
Completion working after adding mise configuration to nushell.

**`mise doctor` output**

n/a

**Additional context**
```
❯ nu --version
0.96.1
```

**Comment by deas** [comment]
Meanwhile, I found a workaround. There may be a more elegant way, but it should probably be emitted by `mise activate nu`.
```nu
def --env "update-env" [] {
  for $var in $in {
    if $var.op == "set" {
      # The following tow lines work around the issue
      let val = if $var.name == "PATH" { $var.value | split row (char esep) } else { $var.value }
      load-env {($var.name): $val}
    } else if $var.op == "hide" {
      hide-env $var.name
    }
  }
}
```

**Comment by last-partizan** [comment]
Yes, this fix works, thanks!

**Comment by last-partizan** [comment]
@jdx it think it's a bug and should be fixed in mise.

I can make a patch with required changes. Or do you think it works as expected?

**Comment by samuelallan72** [comment]
I think this is the same bug as https://github.com/jdx/mise/issues/2319

**Comment by samuelallan72** [comment]
Please see https://github.com/jdx/mise/pull/3592 for a PR with a fix. :)

**Reply by last-partizan** [reply]
Yes, thanks!

**Comment by Marukome0743** [comment]
Hi @deas — closing the loop here: the fix mentioned by @samuelallan72 in #3592 was merged and released.

It implements the same approach as your workaround: `PATH` is handled separately in the generated Nushell activation script and converted from the serialized string back into the list format Nushell expects:

```nu
$env.PATH = ($var.value | split row (char esep))
```

This prevents `mise activate nu` from changing `$env.PATH` into a string and breaking executable completion. The fix has been available since the December 2024 releases, including v2024.12.12.

Could you confirm that the original `qemu-<TAB>` reproduction works with a recent version of mise? Thanks for identifying the cause and proposing the correct fix.

*This comment was generated by an AI coding assistant.*
